src/metabase/server/middleware/session.clj

```diff
@@ -356,10 +356,23 @@
   ;; The amount is nillable.
   (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
   :type       :json
-  :default    nil)
+  :default    nil
+  :setter     (fn [new-value]
+                (when (some? new-value)
+                  ;; Assert the duration is less than 100 years.
+                  ;; Otherwise, we could fail to format the expires date if the year
+                  ;; has more than 4 digits (#25253)
+                  (let [{:keys [unit amount]} new-value
+                        days (.toDays (case unit
+                                        "seconds" (t/seconds amount)
+                                        "minutes" (t/minutes amount)
+                                        "hours"  (t/hours amount)))
+                        years (/ days 365.25)]
+                    (assert (< years 100) "Session timeout must be less than 100 years")))
```

Comment:
Shouldn't this error message be i18n'ed with `tru`? What happens when someone tries to set this to something invalid and isn't using English?

Reply:
Yes thanks, updated https://github.com/metabase/metabase/commit/f7d4b9917c7e564f21604d715c622d8fca60f17e